backend/infrahub/graphql/mutations/main.py

```diff
@@ -189,6 +204,41 @@ async def mutate_create_to_graphql(cls, info: GraphQLResolveInfo, db: InfrahubDa
             result["object"] = await obj.to_graphql(db=db, fields=fields.get("object", {}))
         return cls(**result)
 
+    @classmethod
+    async def _call_mutate_update(
+        cls,
+        info: GraphQLResolveInfo,
+        data: InputObjectType,
+        branch: Branch,
+        db: InfrahubDatabase,
+        obj: Node,
+    ) -> tuple[Node, Self]:
+        """
+        Wrapper around mutate_update to potentially activate locking and call it within a database transaction.
+        """
+
+        lock_names = registry.schema.get_schema_branch(name=branch.name).get_kind_lock_names_on_object_mutation(
```

Comment:
we should get the schema branch from the `InfrahubDatabase` object : `db.get_schema_branch()`